From 29d87dc8f4ea9a410b027bd6dc5ba0160063319a Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Tue, 21 Mar 2023 09:26:04 +0200
Subject: [PATCH] [SW-93113] gaudi2: rom: check value returned from function

This patch fixes an issue pointed by synopsis static analysis tool by
checking the value returned from function dma_wait_finish in
arcboot_rom_load_recovery and take appropriate action.

"47712 Unchecked return value
If the function returns an error value, the error value may be mistaken
for a normal value. In arcboot_rom_load_recovery: Value returned from a
function is not checked for errors before being used (CWE-252)"

Change-Id: I7acb7c53708c3a3f4614877da8e5d522ba245ae4
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 86f82b7..0340f12 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -265,7 +265,8 @@
 	recovery_img_size = spi_flash_readl(0);
 	dma_copy_image(recovery_img_size, 0,
 		HL_GEN_MEM_BASE_FULL + HL_ICCM_OFFSET, SPI_DMA_DST_HBW);
-	dma_wait_finish();
+	if (dma_wait_finish() == DMA_CRC_ERROR)
+		return;
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_RECOVERY_DMA);
 
 	/* jump to iccm start location */
